toolbox/dicom.py

```diff
@@ -97,43 +107,75 @@ def get_elements(self, ds):
                 raise Exception(f"There is a duplicate dicom tag, please check the logic: \n {de.name}: {value}")
         return content
 
-
-class InterestedDicomTagExtractor:
-    tag_of_interested = [
-        'Series Description',
-        'MR Acquisition Type',
-        'Slice Thickness',
-        'Imaged Nucleus',
-        'Number of Averages',
-        'Echo Number(s)',
-        'Magnetic Field Strength',
-        'Spacing Between Slices',
-        'In-plane Phase Encoding Direction',
-        'Patient Position',
-        'Photometric Interpretation',
-        'Rows',
-        'Columns',
-        'Pixel Spacing',
-        'Pixel Representation',  # Pixel Representation (0028,0103) is either unsigned (0) or signed (1)
-        'Window Center',
-        'Window Width',
-        'Rescale Intercept',
-        'Rescale Slope',
-        'Rescale Type',
-    ]
-    keys = tag_of_interested + [SUBJECT_ID, SEQUENCE_NAME]
-
-    @classmethod
-    def get_df(cls, df_dcm_file):
-        dicom_tag_processor = DicomTagProcessor()
-        subject_list = []
-        for _, row in df_dcm_file.iterrows():
-            ds = pydicom.dcmread(row["path"], stop_before_pixels=True)
-            elements = dicom_tag_processor.get_elements(ds)
+    def get_tags_of_interest(self, dicom_file, tags_of_interest):
+        keys, subject_list = tags_of_interest + [SUBJECT_ID, SEQUENCE_NAME], []
+        for _, row in DicomTagProcessor._get_dicom_df(dicom_file).iterrows():
```

Comment:
`DicomTagProcessor._get_dicom_df` -> `self._get_dicom_df`

Reply:
Since `_get_dicom_df `is static, no `self `argument is needed.

Reply:
self._get_dicom_df is more common to call the static class function though